fastlane/lib/fastlane/actions/hockey.rb

```diff
@@ -13,17 +13,20 @@ class HockeyAction < Action
       def self.run(options)
         # Available options: http://support.hockeyapp.net/kb/api/api-versions#upload-version
 
+        @options = options
+
         require 'shenzhen'
         require 'shenzhen/plugins/hockeyapp'
 
         if options[:dsym]
           dsym_filename = options[:dsym]
         else
-          if options[:ipa].to_s.length == 0
-            UI.user_error!("You have to provide an ipa file")
+
+          if build_file.nil?
+            UI.user_error!("You have to provide a build file")
           end
 
-          dsym_path = options[:ipa].gsub('ipa', 'app.dSYM.zip')
+          dsym_path = options[:ipa].to_s.gsub('ipa', 'app.dSYM.zip')
           if File.exist?(dsym_path)
```

Comment:
This will result in `File.exist?("")` if the user didn't provide an `ipa`. Let's change that to 

```
options[:ipa] && File.exist?(dsym_path)
```